### PR TITLE
Subscriptions: Allow user to enable Subscriptions from Subscribe Block and pre-publish panel

### LINKED ISSUE #14621: Subscriptions block: Make it always available
The subscriptions block is only available when the module is enabled, which means users wont find it unless they know where to look.

#### Describe the solution you'd like
We should make the block available for all users and show a nudge to turn on the module.

### LINKED ISSUE #28288: Subscriptions: Add pre-publish UI when feature is disabled, with CTA to enable
### Impacted plugin

Jetpack

### What

It would be helpful to indicate when Subscriptions are disabled in the pre-publish UI

### How

As a Jetpack user, I wasn't aware Subscriptions were disabled when I was publishing my post.

It would be helpful to have some UI indicating such:

image

This way, I know to go turn on Subscriptions before publishing my post, so my subscribers receive my post.

From https://github.com/Automattic/jetpack/pull/28204#issuecomment-1374014990

### PR DESCRIPTION
**Pull request by kozer**
Fixes #14621
Fixes #28288

## Proposed changes:

This pr tries to address two issues when subscriptions are disabled.

1. The subscriptions block is not available when the module is disabled, which means users won't find it unless they know where to look.
2. Users are not aware that Subscriptions are disabled when they are publishing a post.

### Subscription module

We can activate the subscription module using jetpack. When a user visits `Jetpack > Settings` he
can find the Subscriptions module there:

how_to_activate_subscription

### Subscription block is not available when the module is disabled

When the user creates a new post and we try to add a new subscription block, this is not there:

no_module_disabled

If we activate the module the block appears just fine:

block_appears_enabled

#### Solving the issue

In the `subscriptions.php` file, under `/extensions/blocks/subscriptions` there is one function
responsible to make the block accessible in the UI:

[register_block](https://github.com/Automattic/jetpack/pull/29044/files#diff-9164d71f603706b3c6cc58c71ba3f2fbb12b1dba772adca76793f60c24ddbca7L36)

We need to let jetpack register the block and tackle this in the UI by showing a placeholder to the
user to indicate that he must activate the subscriptions first before he can use the module.

##### UseModuleStatus hook

Right now we don't have a way to enable/disable a module from a block.
Luckily for us, there is a `js-package` that plugins can be shared called `Shared Extension Utilities`
In there I have created a new hook that provides the following shared functionalities:

1. Can enable/disable a module
2. Has the current status information of the module we are interested in.

Because a hook is not globally accessible just like a reducer is, and is rendered with a "cleaned
state" whenever we reuse it down in the component tree, and so, there are calls that are re-running,
we can, later on, move this hook logic into a reducer if we find that we use it too much to reduce the API calls.

For our use case, it is ok though, as it's used up to the hierarchy of the component tree, and we
won't notice this issue.

This hook is responsible to get the module status (Subscriptions) and allows us to show a
placeholder when it is appropriate.

##### Subscriptions placeholder

Using the hook mentioned above, we can now display a placeholder to the user that he must activate
the subscriptions first, before he can use the module:

placeholder_activate

### Users are not aware that Subscriptions were disabled

Besides the subscriptions block we see above, there is no information indicating to the user that the
The subscriptions module is disabled when he publishes a new post.

When the user has the module enabled, we will indicate that the post is shared in:

- Pre-publish panel

pre_publish

We can reuse this area to show the user a similar message as we did when he adds a subscription
block.

Using the hook, we achieved that as shown below:

- Pre-publish panel

pre_pub_disabled

### TL;DR
- Hook to update jetpack module status, from #29066.
- Add proper UI to enable subscriptions for:
     - Pre-publish panel
     - Subscription block

Finally, I have added a skeleton UI while the subscription block is loading:

subscripton-block-skeleton-loader

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1676970563872519-slack-CDLH4C1UZ

## Does this pull request change what data or activity we track or use?

Yes, it adds Tracks events around activating the module.

### Subscription block

1. Checkout this branch and make sure that you have jurassic tube enabled.
2. Create a new post.
3. Add a new block.
4. Search for subscription block.
5. Ensure that it's there:

placeholder_activate

6. Press the `Activate Subscriptions` button.
7. Ensure that you now see the subscription block:

subscription_block_active

### Subscription block loader

1. Checkout this branch and make sure that you have jurassic tube enabled.
2. Create a new post.
3. Add a new block.
4. Search for the subscription block.
5. From network tab change to "Slow 3g"
6. Add the block
7. Ensure that you now see the subscription block skeleton loader:

subscripton-block-skeleton-loader

### Jetpack pre publish panel

1. Check out this branch and make sure that you have Jurassic tube enabled.
2. Create a new post.
3. Press publish button
5. Ensure that the `Subscriptions` menu is there, and looks like that:

pre_pub_disabled

6. Press `Activate Subscriptions`
7. Ensure that the `Subscriptions` area has now been changed:

pre_publish

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: IIRC we're trying to avoid using Lodash, so probably better to use a different approach.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:36]: What are the implications of this change?

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: @Automattic/gold Do you have any opinions on what we should display when Newsletters are disabled?

Here's the issue we're trying to solve: https://github.com/Automattic/jetpack/issues/28288

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (kozer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: This line is actually borrowed from another part of the code.
I'll make it simpler and remove the lodash get.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php
Comment (kozer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:36]: This is something, I'm not very curtain about.
When I remove this, it runs the js code, even when the subscription module is not enabled.
However I'm not sure of the following:
- Is there any use case where the subscription is not there at all?

If yes, we need to ensure that the plugin will not load in that case, because the button to enable newsletter will not make any sense.

Also I don't know what this `register_block` does in comparison to js "register block"

Can you guide me on that?

Comment (danielbachhuber) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:36]: > Also I don't know what this `register_block` does in comparison to js "register block"
> 
> Can you guide me on that?

What exploration have you already done?

Comment (kozer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:36]: @danielbachhuber I tried to search in field guide. 
However, I have found Gutenberg docs, and to my understanding, it seems that it registers the block (js needs to use the same name to register it as well).
Is this correct?

Comment (danielbachhuber) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:36]: @kozer What have you found in the code? And how does the user experience differ with this change applied?

Comment (kozer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:36]: This specific part actually, lets the js client render the block.
I will add what was changed and screenshots about those changes tomorrow. 
It's just a draft and I'd like your feedback on if this is a good way to go.

Long story, short, this whole pr adds a proper UI to enable subscriptions at:
- Pre-publish panel
- Post publish panel
- Subscription block

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (artpi) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: >  Do you have any opinions on what we should display when Newsletters are disabled?

Probably some kind of informational text?

Comment (artpi) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: "The newsletter functionality is currently disabled. Enable it to send your posts to your email followers as a newsletter" 
Or something like that CC @mtias

Comment (artpi) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: CC @mtias

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php
Comment (artpi) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:36]: It seems that doing it this way will also solve https://github.com/Automattic/jetpack/issues/14621 so this may be the way to go

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (artpi) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: How will this work on WPCOM sites where this interface is in Calypso?

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: I'm thinking we may want to add a bit of an explanation here, so folks know why it would be good for them to enable this, and what would happen when they do.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: Just like above, I think we would need a bit more incentive to push folks to enable the feature.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js:100]: Should we have a "loading" state, so we can change the look of the button when the module is being activated?

**Review by jeherve — COMMENTED** [review]
Nice work! I think this will contribute to make the Subscriptions feature a lot more discoverable, and that's definitely a plus.

I noted a few things below.

- You will need to make changes to tests, they're failing at the moment.
- We may want to only merge this when #29028 is merged. This will ensure that enabling Subscriptions from the block editor has no side-effect that is unknown to the site owner (namely adding fields below the comment forms, since once #29028 is merged those will not be automatically enabled).
- I wonder if we should maybe not offer a way to activate Subscriptions from the post-publish panel. At this point, the post is already published, however if we activate the subscriptions then, the post will not be sent to subscribers, even though the post-publish panel will say so.
- Since we will be offering a Subscribe block to all site owners, even those that previously used the Subscriptions feature and eventually elected to turn it off in favor of something else. I think we should be ready to acknowledge that in our documentation by offering a way for site owners and potentially hosting partners to disable the block. Something like:
```php
add_filter(
	'jetpack_set_available_extensions',
	function ( $extensions ) {
		return array_diff( $extensions, array( 'subscriptions' ) );
	}
);
```
This may be enough, or we may have to introduce a filter to disable the prompts to enable in the block editor. I'm trying to get clarification on that in p1676995485200439-slack-CQN0YDYM7

**Comment by artpi** [comment]
This issue seems connected https://github.com/Automattic/wp-calypso/issues/72652
CC @millerf

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (kozer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js:100]: I  disabled the button based on the `isLoading` state from the hook.
Isn't that enough?

Comment (jeherve) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js:100]: In some other places, we have a loading state:


https://user-images.githubusercontent.com/426388/220672613-fe193616-9602-4737-810c-59920015771b.mov

https://github.com/Automattic/jetpack/blob/3a3c4adc1c0913bac224ddb832713bd6101fe1ab/projects/plugins/jetpack/extensions/blocks/contact-form/components/jetpack-crm-integration/jetpack-crm-integration-settings-plugin-state.js#L53-L61

Maybe we could have something similar here?

Comment (kozer) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js:100]: > In some other places, we have a loading state:
> 
> https://github.com/Automattic/jetpack/blob/3a3c4adc1c0913bac224ddb832713bd6101fe1ab/projects/plugins/jetpack/extensions/blocks/contact-form/components/jetpack-crm-integration/jetpack-crm-integration-settings-plugin-state.js#L53-L61
> 
> Maybe we could have something similar here?

Oh! Didn't know that! Thank you very much!!

**Comment by millerf** [comment]
> This issue seems connected https://github.com/Automattic/wp-calypso/issues/72652
> CC @millerf

This is some awesome work!
And it looks complementary. The work I have done so far is gated by `isNewsletterFeatureEnabled()`, so it should be safe...

**Comment by kozer** [comment]
Question:

Hello @millerf , @danielbachhuber and @jeherve 

I notice that the hook we introduced (thanks @jeherve for your original idea), as it doesn't share the state between components, creates a visual mismatch in some cases.

Here is an example of that:

visual_mismatch_subscr

Do you think is better to solve this by creating a new store now, or opening an issue and merging this first?

In case we want to tackle this now:

How we can add a context/redux store that is shared between the block and jetpack panel? I mean that we use 2 different places we add things:

```
// Registers Subscribe block.
registerJetpackBlock( name, settings );
```

and

```
// Registers slot/fill panels defined via settings.render.
registerJetpackPlugin( name, pluginSettings );
```

Is there a way to share things between the two?

If it was something like:

```
<App>
  <Block/>
  <Sidebar/>
</App>
```
it would be an easy thing to do:

```
<App>
<Store>
  <Block/>
  <Sidebar/>
</Store/>
</App>
```

but what about now?

Thank you very much!

**NOTE:**

@jeherve regarding your suggestion on adding a loading state, I reused what already exists in the `@wordpress/components` button component. It has an `isBusy` property that adds a loading state to the button if needed. (You can take a look at how this is looked like [here](https://wordpress.github.io/gutenberg/?path=/story/components-button--secondary&args=isBusy:true))

Do you think it is ok, or I should create something similar to what you suggested to match that?

IMO, I think it's better to use something that we already have "universally" , and create consistency across the product, and this is why I used the button component, but I'd love to tell me your thoughts on that.

**NOTE 2**

I see that `continuous-integration/a8c-teamcity — The build is failure. D102612-code` is failing.
But I can't find out why. Can someone guide me through that? 
Thanks!

File: projects/js-packages/shared-extension-utils/changelog/use-module-status
Comment (danielbachhuber) [inline projects/js-packages/shared-extension-utils/changelog/use-module-status]:
```suggestion
useModuleStatus: Add new hook to enable or disable Jetpack modules.
```

File: projects/js-packages/shared-extension-utils/src/hooks/use-module-status.js
Comment (danielbachhuber) [inline projects/js-packages/shared-extension-utils/src/hooks/use-module-status.js]: Should we add some tests for this hook?

File: projects/plugins/jetpack/changelog/update-subscriptions-inactive
Comment (danielbachhuber) [inline projects/plugins/jetpack/changelog/update-subscriptions-inactive]:
```suggestion
Subscriptions: Makes the Subscribe Block available, even when the Subscriptions module is off, so Subscriptions can be easily enabled.
```

File: projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js]: Can we use a skeleton screen placeholder state while this data is loading? It appears as broken with zero-height (and causes layout shift when it loads):

https://user-images.githubusercontent.com/36432/221225404-ca2f3252-8e7a-4c32-ab79-6eba7ab040d1.mp4

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]:
```suggestion
							{ isChangingStatus ? __( 'Activating Subscriptions', 'jetpack' ) : __( 'Activate Subscriptions', 'jetpack' ) }
```

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]: I don't think we need to show 'Activate Subscriptions' in the post-publish panel. As @jeherve pointed out, doing so opens the door for a confusing user experience: activating Subscriptions won't send the already-published post.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscription-placeholder.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscription-placeholder.js]:
```suggestion
					{ __( 'Learn more about the Subscriptions feature.', 'jetpack' ) }
```

File: projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js]:
```suggestion
					{ __( 'Learn more about the Subscriptions feature.', 'jetpack' ) }
```

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscription-placeholder.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscription-placeholder.js]:
```suggestion
				{ isLoading ? __( 'Activating Subscriptions', 'jetpack' ) : __( 'Activate Subscriptions', 'jetpack' ) }
```

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:55]: Nice!

File: projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php:361]: Is it easily possible to add a test for this? If it is, it would be nice to add one.

File: projects/plugins/jetpack/extensions/blocks/subscriptions/test/edit.js
Comment (danielbachhuber) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/test/edit.js:117]: Nice! Thanks for adding these tests.

**Review by danielbachhuber — COMMENTED** [review]
Great work on this so far!

I tested by creating a jurassic.ninja site, installing the Jetpack beta tester plugin, and checking out the `feature/subscriptions_pre_publish_ui` branch.

Here's how it looked to enable Subscriptions from the Subscribe block:

https://user-images.githubusercontent.com/36432/221228373-519bafce-5a4d-4f44-a167-5013a6d5be4f.mp4

I also ran into the data disconnect state you described:

CleanShot 2023-02-24 at 07 49 54@2x

Given both UI are a part of the same block JavaScript, maybe we could store state in the parent caller?

I left a few comments on smaller things. Looks like this is getting close to being shippable 😊

**Comment by kozer** [comment]
> Given both UI are a part of the same block JavaScript, maybe we could store state in the parent caller?

This is what I'm not getting. One is part of the `panel.js` file and the other is part of `edit.js` file.
Both of those files are connected through `index.js` file, which is a js object that `editor.js` file uses and does the following:

```
// Registers Subscribe block.
registerJetpackBlock( name, settings );

// Registers slot/fill panels defined via settings.render.
registerJetpackPlugin( name, pluginSettings );
```

From the above lines, I understand that block and panel are two different things. How I can use the hook to pass the state down to those two "components"? Where is the place we can do that?

**Comment by jeherve** [comment]
> I notice that the hook we introduced (thanks @jeherve for your original idea), as it doesn't share the state between components, creates a visual mismatch in some cases.
>
> Do you think is better to solve this by creating a new store now, or opening an issue and merging this first?

I would personally vote for splitting the work. I would aim to merge this first parter and iterate on this in follow-up PRs.
We could have a first goal of not displaying the pre- and post-publish panels at all when the module isn't active in this PR. In a follow-up PR, once we have the store set up, we can start displaying the publish panels with that in mind.

That may mean rolling back some of your panel changes from this PR, but we can keep them around for the follow-up PR, that's not all lost.

> From the above lines, I understand that block and panel are two different things. How I can use the hook to pass the state down to those two "components"? Where is the place we can do that?

I'll let @danielbachhuber weigh in on that, but I wonder if it may be worth taking a more general approach, not limited to subscriptions, to get and manage module status in any Jetpack block. The hook we introduce isn't limited to subscriptions, and there are other blocks that will directly benefit from this and could also be made more discoverable. #10903 is one such example, and there are other blocks just like that, relying on a module.

This may mean introducing a store alongside the hook, and rely on it directly inside `registerJetpackBlock` and `registerJetpackPlugin`?

> Regarding your suggestion on adding a loading state, I reused what already exists in the `@wordpress/components` button component. It has an `isBusy` property that adds a loading state to the button if needed.

That's perfect! 👍

File: pnpm-lock.yaml
Comment (danielbachhuber) [inline pnpm-lock.yaml:790]: @jeherve Is it fine to add this?

File: projects/js-packages/shared-extension-utils/package.json
Comment (danielbachhuber) [inline projects/js-packages/shared-extension-utils/package.json:3]: @jeherve What do we need to know about tagging a new release of this package? Should we take responsibility for tagging the release, or will someone else?

Comment (jeherve) [reply projects/js-packages/shared-extension-utils/package.json:3]: A new version of the package will be automatically released next time a plugin that uses it needs to be released. In practice, that most likely means the next Jetpack weekly release.

In general, we don't need to worry about package releases; Ground Control takes care of them as they release plugins. We only need to make sure the dev version number is correct and matches the changes in the monorepo (like here; we're introducing new changes classified as minor to the package, so the version number must be bumped to a new minor version).

File: pnpm-lock.yaml
Comment (jeherve) [reply pnpm-lock.yaml:790]: Yup, no issues here. We can add that new dependency to the package, and when doing so we end up having to modify the workspace's lock file.

**Comment by danielbachhuber** [comment]
@kozer Can you resolve the merge conflict (and other warnings) and then I'll give it another review?

**Review by jeherve — COMMENTED** [review]
I pushed a few commits to update with latest trunk, solve conflicts with the pnpm.lock file, and fix an annoying build failure that's caused by translations in ternaries (see b2d76def52504b7ab45d11342f7f230d97f4ab5d).

It should build well now. It does work well overall, nice work!

I notice an error when activating the module from the Subscribe block; the block tries to make a request to the `wpcom/v2/memberships/product` endpoint right away (see https://github.com/Automattic/jetpack/blob/a572b28ca6f091a284d1d5ff419917682d28add9/projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js#L325-L332 ), but the endpoint isn't available yet since we haven't reloaded the page. I'm not sure how to best address that in this case.

**Comment by danielbachhuber** [comment]
Whomp, another merge conflict.

**Comment by kozer** [comment]
> Whomp, another merge conflict.

Thanks for noticing! Fixed!

**Comment by kozer** [comment]
> I pushed a few commits to update with latest trunk, solve conflicts with the pnpm.lock file, and fix an annoying build failure that's caused by translations in ternaries (see [b2d76de](https://github.com/Automattic/jetpack/commit/b2d76def52504b7ab45d11342f7f230d97f4ab5d)).
> 
> It should build well now. It does work well overall, nice work!
> 
> I notice an error when activating the module from the Subscribe block; the block tries to make a request to the `wpcom/v2/memberships/product` endpoint right away (see
> 
> https://github.com/Automattic/jetpack/blob/a572b28ca6f091a284d1d5ff419917682d28add9/projects/plugins/jetpack/extensions/blocks/subscriptions/edit.js#L325-L332
> 
> ), but the endpoint isn't available yet since we haven't reloaded the page. I'm not sure how to best address that in this case.

I investigated a bit, and couldn't find a solid way either. Force reloading could be an option, or prompt the user in the UI to do so, however, I don't find this is a good solution from a UI/UX perceptive. How do you think I should proceed here, based on your experience?

**Comment by jeherve** [comment]
> I don't find this is a good solution from a UI/UX perceptive. How do you think I should proceed here, based on your experience?

I'm not sure, to be honest. @artpi Do you have an opinion on this, since you worked on the Paid Newsletters UI?

**Comment by kozer** [comment]
@artpi, @jeherve , any news/ideas for this pr? I keep updating it, however, I'm not sure how to proceed from here.

**Review by jeherve — COMMENTED** [review]
Looks like you'll need to update the tests as well:

```
 1) WP_Test_Jetpack_Subscriptions::test_comments_are_displaying_on_not_accessible_pages
  Failed asserting that true is false.
  
  /tmp/wordpress-trunk/src/wp-content/plugins/jetpack/tests/php/modules/subscriptions/test_class.jetpack-subscriptions.php:367
```

**Comment by kozer** [comment]
> Looks like you'll need to update the tests as well:
> 
> ```
>  1) WP_Test_Jetpack_Subscriptions::test_comments_are_displaying_on_not_accessible_pages
>   Failed asserting that true is false.
>   
>   /tmp/wordpress-trunk/src/wp-content/plugins/jetpack/tests/php/modules/subscriptions/test_class.jetpack-subscriptions.php:367
> ```

Thanks @jeherve ! Fixed!

File: projects/plugins/jetpack/tests/php/modules/subscriptions/test_class.jetpack-subscriptions.php
Comment (jeherve) [inline projects/plugins/jetpack/tests/php/modules/subscriptions/test_class.jetpack-subscriptions.php:359]: Do we need to do some cleanup to remove the filter after the test, to avoid any kind of pollution this test may leave behind with the filter?

**Comment by wojtekn** [comment]
> I notice an error when activating the module from the Subscribe block; the block tries to make a request to the wpcom/v2/memberships/product endpoint right away, but the endpoint isn't available yet since we haven't reloaded the page. I'm not sure how to best address that in this case.

@jeherve I tried reproducing this, but with no luck. I used local Jetpack with Jurassic Tube and:

1. Ensured that subscriptions are disabled in Jetpack -> Settings -> Discussion
2. Opened post editor
3. Added "Subscribe" block
4. Clicked a button to activate Subscriptions
5. In the end, Subscribe block appeared correctly in the editor without any errors in the console. I published the post and confirmed that the Subscribe block is displayed in the frontend.

@kozer a note to test instructions - after checking out the latest branch I also needed to build the plugin:

`jetpack build plugins jetpack`

**Comment by jeherve** [comment]
> @jeherve I tried reproducing this, but with no luck. I used local Jetpack with Jurassic Tube and:

That's a good point! It's limited to sites where the new Paid Newsletter is enabled. So far it hasn't been enabled on most sites (only proxied a12s have access to the feature on WordPress.com Simple right now, for example); you would need to force-enable it like so:

```php
add_filter( 'jetpack_subscriptions_newsletter_feature_enabled', '__return_true' );
```

**Review by jeherve — DISMISSED** [review]
This tests well for me overall. Thanks for adding those tests alongside your changes.

I think we could merge this as is. There are still some questions / issues pending as we discussed above, but those could be addressed in a follow-up PR.

File: projects/plugins/jetpack/tests/php/modules/subscriptions/test_class.jetpack-subscriptions.php
Comment (kozer) [reply projects/plugins/jetpack/tests/php/modules/subscriptions/test_class.jetpack-subscriptions.php:359]: Isn't that filter only available under this test? (No idea, just asking, I'll clear this filter though)

**Review by wojtekn — DISMISSED** [review]

**Review by wojtekn — APPROVED** [review]

**Review by jeherve — APPROVED** [review]

**Comment by danielbachhuber** [comment]
Way to see this one through, @kozer !

**Pull request merged**
